target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-wia3300-20.dts

```diff
@@ -0,0 +1,447 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+
+/ {
+	compatible = "skspruce,wia3300-20", "qcom,ipq4019";
+	model = "SKSpruce WIA3300-20";
+
+	aliases {
+		label-mac-device = &gmac;
+		led-boot = &led_status_red;
+		led-failsafe = &led_status_lime;
+		led-running = &led_status_lime;
+		led-upgrade = &led_status_red;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		pinctrl-0 = <&key_pins>;
+		pinctrl-names = "default";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 18 (GPIO_ACTIVE_LOW|GPIO_PULL_DISABLE)>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		pinctrl-0 = <&led_pins>;
+		pinctrl-names = "default";
+
+		led_status_red: led-0 {
+			color = <LED_COLOR_ID_RED>;
+			function = LED_FUNCTION_STATUS;
+			gpios = <&tlmm 8 (GPIO_ACTIVE_LOW|GPIO_OPEN_DRAIN|GPIO_PULL_DISABLE)>;
+		};
+
+		led_status_lime: led-1 {
+			color = <LED_COLOR_ID_LIME>;
+			function = LED_FUNCTION_STATUS;
+			gpios = <&tlmm 9 (GPIO_ACTIVE_LOW|GPIO_OPEN_DRAIN|GPIO_PULL_DISABLE)>;
+		};
+
+		led-2 {
+			color = <LED_COLOR_ID_LIME>;
+			function = LED_FUNCTION_WLAN_2GHZ;
+			gpios = <&tlmm 11 (GPIO_ACTIVE_LOW|GPIO_OPEN_DRAIN|GPIO_PULL_DISABLE)>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led-3 {
+			color = <LED_COLOR_ID_LIME>;
+			function = LED_FUNCTION_WLAN_5GHZ;
+			gpios = <&tlmm 28 (GPIO_ACTIVE_LOW|GPIO_OPEN_DRAIN|GPIO_PULL_DISABLE)>;
+			linux,default-trigger = "phy1tpt";
+		};
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x80000000 0x10000000>;
```

Comment:
Is it really neccessary, usually bootloader will patch it in

Reply:
Removed. Yes I can still see this node in device tree after removing it.
```
root@OpenWrt:/sys/firmware/devicetree/base/memory# free
              total        used        free      shared  buff/cache   available
Mem:         246160       85900      137716         388       22544      124216
Swap:             0           0           0
root@OpenWrt:/sys/firmware/devicetree/base/memory# xxd reg
00000000: 8000 0000 1000 0000
```